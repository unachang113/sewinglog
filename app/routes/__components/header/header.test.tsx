import {render} from '@testing-library/react';
import {createRemixStub} from '@remix-run/testing';
import {Header} from '.';

describe('Header', () => {
  const RemixStub = createRemixStub([
    {
      path: '/',
      id: 'posts',
      Component: () => <Header />,
    },
  ]);

  it('HTMLとスタイルが変わらないこと', async () => {
    const {asFragment} = render(<RemixStub />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Sewing logというタイトルが表示されていること', () => {
    const {getByRole} = render(<RemixStub />);

    expect(getByRole('heading', {level: 1, name: 'Sewing log'})).toBeInTheDocument();
  });

  it('linkのpathが/で指定されていること', () => {
    const {getByRole} = render(<RemixStub initialEntries={['/']} />);
    expect(getByRole('link')).toHaveAttribute('href', '/');
  });
});
