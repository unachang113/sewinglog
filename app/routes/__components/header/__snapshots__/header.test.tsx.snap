// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > HTMLとスタイルが変わらないこと 1`] = `
<DocumentFragment>
  <header
    class="px-4 py-4 w-full border-b-2 border-primary bg-white"
  >
    <h1
      class="font-bold text-2xl text-center font-logo text-primary"
    >
      <a
        data-discover="true"
        href="/"
      >
        Sewing log
      </a>
    </h1>
  </header>
</DocumentFragment>
`;
